src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexQueryingScope.cs

```diff
@@ -91,6 +91,9 @@ public void RecordAlreadyPagedItemsInPreviousPage(TopDocs search, CancellationTo
             if (search.ScoreDocs.Length <= _alreadyScannedForDuplicates)
                 return;
 
+            if (search.ScoreDocs.Length <= _query.Start)
```

Comment:
Please add in the test a case that `search.ScoreDocs.Length == _query.Start` so it will be verified too

Reply:
done